pkg/pgmodel/model/pgsafetype/custom_pgtypes.go

```diff
@@ -0,0 +1,99 @@
+package pgsafetype
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/jackc/pgtype"
+	"github.com/timescale/promscale/pkg/pgmodel/model"
+)
+
+// Text is a custom pgtype that wraps the pgtype.Text. It is safe from null characters.
+type Text struct {
+	pgtype.Text
+}
+
+func (t *Text) Set(src interface{}) error {
+	text := src.(string)
+	if err := t.Text.Set(sanitizeNullChars(text)); err != nil {
+		return fmt.Errorf("safe-text: %w", err)
+	}
+	return nil
+}
+
+func (t *Text) Get() interface{} {
+	return revertSanitization(t.Text.Get().(string))
+}
+
+func (t *Text) AssignTo(dst interface{}) error {
+	if err := t.Text.AssignTo(dst); err != nil {
+		return fmt.Errorf("safe-text: %w", err)
+	}
+	return nil
+}
+
+func (t *Text) Scan(src interface{}) error {
```

Comment:
don't we need to sanitizeNullChars here?

Reply:
Does seem so. The end-to-end tests speak that the current implementation is sufficient to handle this stuff. I will implement some unit tests to make this more easily noticeable/correctable/testable.